tests/Elewant/Webapp/Application/Controllers/ApiCommandBase.php

```diff
@@ -29,174 +30,144 @@ abstract class ApiCommandBase extends WebTestCase
-        );
-
-        return $client;
-    }
-
-    private function client()
-    {
-        $client = static::createClient();
-
-        if ($client->getContainer() === null) {
-            throw new \RuntimeException('Kernel has been shutdown or not started yet.');
-        }
-
-        $this->getStore($client->getContainer())->attach(
-            'appendTo',
-            function (ActionEvent $event): void {
-                foreach ($event->getParam('streamEvents', new \ArrayIterator()) as $recordedEvent) {
-                    $this->recordedEvents[] = $recordedEvent;
-                }
-            }
-        );
-
-        return $client;
-    }
-
-    protected function retrieveHerdFromListing($herdId)
+    /**
+     * @param string $herdId
+     * @return mixed[]|null
+     */
+    protected function retrieveHerdFromListing(string $herdId): ?array
     {
         $herdListing = $this->getHerdListing($this->client()->getContainer());
 
         return $herdListing->findById($herdId);
     }
 
-    protected function retrieveElePHPantFromListing($elePHPantId)
+    /**
+     * @param string $elePHPantId
+     * @return mixed[]|null
+     */
+    protected function retrieveElePHPantFromListing(string $elePHPantId): ?array
     {
         $herdListing = $this->getHerdListing($this->client()->getContainer());
 
         return $herdListing->findElePHPantByElePHPantId($elePHPantId);
     }
 
-    protected function retrieveHerdElePHPantsFromListing($herdId)
+    /**
+     * @param string $herdId
+     * @return mixed[]
+     */
+    protected function retrieveHerdElePHPantsFromListing(string $herdId): array
     {
         $herdListing = $this->getHerdListing($this->client()->getContainer());
 
         return $herdListing->findElePHPantsByHerdId($herdId);
     }
 
-    protected function retrieveDesiredBreedsFromListing($herdId)
+    /**
+     * @param string $herdId
+     * @return mixed[]
+     */
+    protected function retrieveDesiredBreedsFromListing(string $herdId): array
     {
-
         $herdListing = $this->getHerdListing($this->client()->getContainer());
 
         return $herdListing->findDesiredBreedsByHerdId($herdId);
     }
 
-    /**
-     * @param ContainerInterface $container
-     *
-     * @return ActionEventEmitterEventStore
-     */
-    private function getStore(ContainerInterface $container)
+    private function getStore(ContainerInterface $container): ActionEventEmitterEventStore
     {
         return $container->get('prooph_event_store.herd_store');
     }
 
-    /**
-     * @param ContainerInterface $container
-     *
-     * @return HerdListing
-     */
-    private function getHerdListing(ContainerInterface $container)
+    private function getHerdListing(ContainerInterface $container): HerdListing
     {
         return $container->get('Elewant\Webapp\Infrastructure\ProophProjections\HerdListing');
     }
 
     /**
-     * This method can run a projection once
+     * This method can run a projection once.
```

Comment:
Such improve. Much punctuation. So better!

Reply:
Oh yeah, this one was uncalled for...

When I wrote the comment I assumed it was another code-style "rule" that there must be a dot after every sentence; but it was probably just a manual addition without an actual improvement in mind?